scripts/stackstorm.js

```diff
@@ -126,16 +124,30 @@ module.exports = function(robot) {
     };
   }
 
-  var api = st2client(opts);
+  var api_client = st2client(opts);
 
   if (env.ST2_API_KEY) {
-    api.setKey({ key: env.ST2_API_KEY });
+    api_client.setKey({ key: env.ST2_API_KEY });
   } else if (env.ST2_AUTH_TOKEN) {
-    api.setToken({ token: env.ST2_AUTH_TOKEN });
+    api_client.setToken({ token: env.ST2_AUTH_TOKEN });
+  }
+
+  function logErrorAndExit(err, res) {
+    if (err) {
+      robot.logger.error(err.stack);
```

Comment:
Did you notice how this line prints the following:
```
ERROR undefined
```
as hubot log on most of the errors?

@jinpingh let's please fix this.

Reply:
Sure, will check err.stack before log it

Reply:
Recent change still makes little sense as it just hides an issue.

`err.stack` is not available in most of the error conditions we've seen.
But we want error reporting to actually work and report.

Please provide an alternate solution to show some useful information as error log if `err.stack` is not defined.

Reply:
For current code, all log err messages and extra information are printed before  call `logErrorAndExit`, this is why did not put extra information here.

Reply:
So based on that, should developer always explicitly log the error before actually calling `logErrorAndExit`? Why error is not logged from the method itself if it's naming is `logError...`?

This puts a lot of knowledge burden on someone to who uses `logErrorAndExit` method and its corner cases and limitations. 
 
Additionally, remember that `logErrorAndExit` could be called from any unhandled exception because we're using it in `robot.error()`.

---

With that, I think we have enough reasons to actually _log_ error from the `logErrorAndExit` method.

Reply:
I removed `logErrorAndExit` entirely, since it didn't fit the semantics of what we wanted, so this conversation is obviated/resolved.